[AOTInductor] Add input information for Triton Kernels in AOTI

muchulee8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #160380
* #163593

Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="459" height="582" alt="Screenshot 2025-09-19 at 10 45 10 AM" src="https://github.com/user-attachments/assets/baa0c251-86e9-49ca-8c6c-fcd2619f7f48" />


After:
<img width="473" height="1130" alt="Screenshot 2025-09-19 at 10 44 53 AM" src="https://github.com/user-attachments/assets/b7942d84-0362-4b9e-9232-14de92bbdd00" />


Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @amjames @chauhang @aakhundov @coconutruben

muchulee8 committed:
[AOTInductor] Add input information for Triton Kernels in AOTI

Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="402" height="523" alt="Screenshot 2025-08-11 at 4 06 14 PM" src="https://github.com/user-attachments/assets/4205a8b6-28cf-4e09-8c65-c87ca0e0d9cb" />

After:
<img width="419" height="864" alt="Screenshot 2025-08-11 at 4 07 01 PM" src="https://github.com/user-attachments/assets/aa19e4a8-4e0a-4e9d-a626-caf2efcfb620" />

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="402" height="523" alt="Screenshot 2025-08-11 at 4 06 14 PM" src="https://github.com/user-attachments/assets/4205a8b6-28cf-4e09-8c65-c87ca0e0d9cb" />

After:
<img width="419" height="864" alt="Screenshot 2025-08-11 at 4 07 01 PM" src="https://github.com/user-attachments/assets/aa19e4a8-4e0a-4e9d-a626-caf2efcfb620" />

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="402" height="523" alt="Screenshot 2025-08-11 at 4 06 14 PM" src="https://github.com/user-attachments/assets/4205a8b6-28cf-4e09-8c65-c87ca0e0d9cb" />

After:
<img width="419" height="864" alt="Screenshot 2025-08-11 at 4 07 01 PM" src="https://github.com/user-attachments/assets/aa19e4a8-4e0a-4e9d-a626-caf2efcfb620" />

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="402" height="523" alt="Screenshot 2025-08-11 at 4 06 14 PM" src="https://github.com/user-attachments/assets/4205a8b6-28cf-4e09-8c65-c87ca0e0d9cb" />

After:
<img width="419" height="864" alt="Screenshot 2025-08-11 at 4 07 01 PM" src="https://github.com/user-attachments/assets/aa19e4a8-4e0a-4e9d-a626-caf2efcfb620" />

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="402" height="523" alt="Screenshot 2025-08-11 at 4 06 14 PM" src="https://github.com/user-attachments/assets/4205a8b6-28cf-4e09-8c65-c87ca0e0d9cb" />

After:
<img width="419" height="864" alt="Screenshot 2025-08-11 at 4 07 01 PM" src="https://github.com/user-attachments/assets/aa19e4a8-4e0a-4e9d-a626-caf2efcfb620" />

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

sraikund16 reviewed (commented):
Can we add unit tests to make sure this works for most cases?

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="459" height="582" alt="Screenshot 2025-09-19 at 10 45 10 AM" src="https://github.com/user-attachments/assets/baa0c251-86e9-49ca-8c6c-fcd2619f7f48" />


After:
<img width="473" height="1130" alt="Screenshot 2025-09-19 at 10 44 53 AM" src="https://github.com/user-attachments/assets/b7942d84-0362-4b9e-9232-14de92bbdd00" />


Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="459" height="582" alt="Screenshot 2025-09-19 at 10 45 10 AM" src="https://github.com/user-attachments/assets/baa0c251-86e9-49ca-8c6c-fcd2619f7f48" />


After:
<img width="473" height="1130" alt="Screenshot 2025-09-19 at 10 44 53 AM" src="https://github.com/user-attachments/assets/b7942d84-0362-4b9e-9232-14de92bbdd00" />


Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 commented:
@pytorchbot merge

muchulee8 committed:
Update on "[AOTInductor] Add input information for Triton Kernels in AOTI"


Summary:
We use record_function to pass in input information to let Kineto show
input information.

Test Plan:
Before:
<img width="459" height="582" alt="Screenshot 2025-09-19 at 10 45 10 AM" src="https://github.com/user-attachments/assets/baa0c251-86e9-49ca-8c6c-fcd2619f7f48" />


After:
<img width="473" height="1130" alt="Screenshot 2025-09-19 at 10 44 53 AM" src="https://github.com/user-attachments/assets/b7942d84-0362-4b9e-9232-14de92bbdd00" />


Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

muchulee8 commented:
@pytorchbot merge
